e2e/pages/adf/dataTablePage.ts

```diff
@@ -44,20 +45,107 @@ export class DataTablePage {
+        return this.getRowsNameWithRoot(content).element(rowByRowName);
+    }
+
+    getColumnValueForRow(file, columnName) {
+        let row = this.getRowByRowName(file);
+        Util.waitUntilElementIsVisible(row);
+        let rowColumn = row.element(by.css(`div[title="${columnName}"] span`));
+        Util.waitUntilElementIsVisible(rowColumn);
+        return rowColumn.getText();
+    }
+
+    async getStyleValueForRowText(rowName, styleName) {
+        let row = element(by.css(`div.adf-data-table-cell[filename="${rowName}"] span.adf-datatable-cell-value[title="${rowName}"]`));
+        Util.waitUntilElementIsVisible(row);
+        return row.getCssValue(styleName);
+    }
+
+    /**
+     * Check the list is sorted.
+     *
+     * @param sortOrder: 'true' if the list is expected to be sorted ascendant and 'false' for descendant
+     * @param locator: locator for column
+     * @return 'true' if the list is sorted as expected and 'false' if it isn't
+     */
+    checkListIsSorted(sortOrder, locator) {
+        let deferred = protractor.promise.defer();
+        let column = element.all(by.css(`div[class*="adf-data-table-cell"][title="${locator}"] span`));
+        Util.waitUntilElementIsVisible(column.first());
+        let initialList = [];
+        column.each(function (currentElement) {
+            currentElement.getText().then(function (text) {
+                initialList.push(text);
+            });
+        }).then(function () {
+            let sortedList = initialList;
+            sortedList = sortedList.sort();
+            if (sortOrder === false) {
+                sortedList = sortedList.reverse();
+            }
+            deferred.fulfill(initialList.toString() === sortedList.toString());
+        });
+        return deferred.promise;
+    }
+
+    rightClickOnRowNamed(rowName) {
+        let row = this.getRowByRowName(rowName);
+        browser.actions().click(row, protractor.Button.RIGHT).perform();
+        Util.waitUntilElementIsVisible(element(by.id('adf-context-menu-content')));
+    }
+
+    clickRowToSelect(rowName) {
+        let row = this.getRowByRowName(rowName);
+        Util.waitUntilElementIsClickable(row);
+        browser.actions().keyDown(protractor.Key.COMMAND).click(row).perform();
+        this.checkRowIsSelected(rowName);
+        return this;
+    }
+
+    getColumnLocator(column) {
+        return by.css(`div[class*='adf-datatable-row'] div[title='${column}'] span`);
+    }
+
+    getTooltip(nodeName) {
+        return this.getRowByRowName(nodeName).element(by.css(`span[title="${nodeName}"]`)).getAttribute('title');
+    }
+
     getFileHyperlink(filename) {
         return element(by.cssContainingText('adf-name-column[class*="adf-datatable-link"] span', filename));
     }
 
     getAllDisplayedRows() {
-        return element.all(this.rows).count();
+        return this.rootElement.all(this.rows).count();
     }
 
     getAllRowsNameColumn() {
```

Comment:
This should be moved to demo-shell page. We shouldn't have specific column names here(e.g. Display name)